packages/twenty-front/src/modules/workflow/components/Workflow.tsx

```diff
@@ -36,7 +36,7 @@ export const Workflow = ({
 
   return (
     <>
-      <WorkflowShowPageEffect workflowId={workflowId} />
+      <WorkflowEffect workflowId={workflowId} />
```

Comment:
I agree that `Workflow` is not an excellent component name and can be confused with the type. I haven't found a good name yet for "the component that wraps the workflow diagram, the effect component that fetches workflow's data, and additional stuff like the workflow status badge."
Why not for `WorkflowContainer`? However, I don't feel it conveys much meaning. I often call this piece of UI "the Workflow Visualizer" verbally. What do you think? `Visualizer > Canvas > Diagram`

For the `WorkflowDiagram`, it's the UI component that displays the diagram (diagram = UI nodes + UI edges) and the data (nodes + edges objects). Metonymies can arrive quickly. I'm open to any better name. `WorkflowDiagramInput`? `WorkflowDiagramDefinition`?

If the name of effect components should be constructed from where they are placed, `WorkflowDiagramCanvasEffect` is undoubtedly correct.
The effect does two things: 1. it sets the current view workflow id to a Recoil state, and 2. it computes the data displayed in the workflow diagram. These data are not tied to the `WorkflowDiagramCanvas` and are used elsewhere, like in the `RightDrawerWorkflow*` components.

Reply:
1. thanks for the clarification about Diagram + Nodes + Edges, let's keep this scoped to the visualization and I recommend we prefix with Diagram every time to avoid confusion: DiagramNode, DiagramEdge, ... as node and edge are quite generic naming (used in graphql for example). 

2. I love WorkflowVizualizer

3. It means you probably need to Effect, one beside the Diagram and one beside the vizualizer :)